### DISCUSSION TITLE
AC1-5.0-E-G2 Illegal Data Address exception response

### DISCUSSION THREAD

**Discussion by markheppell** [nathanmarlor/foxess_modbus#715] [Support] [answered]
I have just had my inverter (AC1-5.0-E-G2) firmware updated and I am still getting an Illegal Data Address exception response

If I do a Home Assistant Developer action of Read registers with a count of 45 I get the following

values:
  "31000": 10
  "31001": 0
  "31002": 0
  "31003": 10
  "31004": 0
  "31005": 0
  "31006": 2497
  "31007": 26
  "31008": 661
  "31009": 5002
  "31010": 2508
  "31011": 10
  "31012": 661
  "31013": 5003
  "31014": 65535
  "31015": 4
  "31016": 670
  "31017": 0
  "31018": 283
  "31019": 380
  "31020": 1956
  "31021": 35
  "31022": 686
  "31023": 263
  "31024": 33
  "31025": 200
  "31026": 400
  "31027": 3
  "31028": 1
  "31029": 1
  "31030": 0
  "31031": 0
  "31032": 0
  "31033": 0
  "31034": 0
  "31035": 0
  "31036": 0
  "31037": 0
  "31038": 0
  "31039": 0
  "31040": 0
  "31041": 0
  "31042": 0
  "31043": 0
  "31044": 0

So it looks like the Elfin is passing data, but if I increase the count to 46 I get

error: >-
  Error reading registers. Type: RegisterType.INPUT; start: 31000; count: 46;
  slave: 247 from tcp://192.168.0.227:502: Exception Response(132, 4,
  IllegalAddress)

Home Assistant debug logs are returning

2024-12-27 16:24:45.452 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Reading addresses on tcp://192.168.0.227:502 247: (10016, 3)
2024-12-27 16:24:45.542 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Modbus error when polling tcp://192.168.0.227:502 247: Exception Response(132, 4, IllegalAddress)

I've kind of reached my knowledge limit and would love to receive some help or guidance as to what to look at next.

Many Thanks
Mark

**Comment by FozzieUK** [comment]
The integration is failing to read the Master, Slave, Manager firmware versions at 10016/17/18 - so it looks like the AC1-G2 may have changed the register map from the old AC1 (G1) input register maps.

In HA if you look in Devices & Services, find the Foxess_Modbus integration and click '1 Device' - what (if any) sensors do you see populated.


To start off can you use Developer action to firstly prove the H1/AC1 input range is invalid

Start Address: 10016, Count: 3, Input - and see what the response is (will likely fail)

Then try

Start Address: 30016, Count: 3, Holding - and see what the response is, paste it here if you get a valid response

If you could also try

Start Address: 10000, Count: 8, Input - and see what the response is

Start Address: 30000, Count: 16, Holding - and see what the response is

Again if you could list the responses here please.

**Comment by markheppell** [comment]
I have 1 device, 89 entities, 13 disabled, 75 unavailable, and 1 that is showing something, which is the Connection Status which reads:-

Error reading registers. Type: RegisterType.INPUT; start: 10016; count: 3; slave: 247 from tcp://192.168.0.227:502: Exception Response(132, 4, IllegalAddress)

![Modbus1](https://github.com/user-attachments/assets/95e50a8f-ac1d-4e27-a8d9-596b0c3b7163)

![Modbus2](https://github.com/user-attachments/assets/723bed4b-5360-4adb-9449-cc3cb2d03596)

![Modbus3](https://github.com/user-attachments/assets/557cdb35-c9c9-46d9-a310-88b310715171)

![Modbus4](https://github.com/user-attachments/assets/1ebdba84-00a5-44cf-96fe-048d906a8975)

**Comment by FozzieUK** [comment]
ok thanks, clearly not responding to some of the input registers but ok on the holding registers.

Inverter Model is in hex 41,43,31,2D,35,2E,30,2D,45,2D,47,32 = 'AC1-5.0-E-G2'

Could do with a quick test to see whether the register map is more akin to the H1_G1 using holding regs or the H1_G2

Can you try these and record the responses please

BMS Connect State
Start Address: 31029, Count: 1, Holding
or
Start Address: 31028, Count: 1, Holding

Inverter State
Start Address: 31027, Count: 1, Holding
or
Start Address: 31028, Count: 1, Holding
or
Start Address: 39063, Count: 1, Holding
Start Address: 39065, Count: 1, Holding

**Comment by markheppell** [comment]
Start Address: 31029, Count: 1, Holding
values:
  "31029": 1
Start Address: 31028, Count: 1, Holding
values:
  "31028": 1
Start Address: 31027, Count: 1, Holding
values:
  "31027": 3
Start Address: 31028, Count: 1, Holding
values:
  "31028": 1
Start Address: 39063, Count: 1, Holding
values:
  "39063": 4
Start Address: 39065, Count: 1, Holding
values:
  "39065": 0

**Reply by FozzieUK** [reply]
Hmmmm bit of both... 

Ok as a quick test can you open file editor and open this file `/homeassistant/custom_components/foxess_modbus/inverter_profiles.py`

go to line 221and edit it change RegisterType from INPUT to HOLDING

i.e.

```
        InverterModelProfile(InverterModel.AC1, r"^AC1-([\d\.]+)", capacity_parser=CapacityParser.H1)
        .add_connection_type(
            Inv.H1_G1,
            ConnectionType.AUX,
            RegisterType.INPUT,
            special_registers=H1_AC1_REGISTERS,
        )
```

becomes

```
        InverterModelProfile(InverterModel.AC1, r"^AC1-([\d\.]+)", capacity_parser=CapacityParser.H1)
        .add_connection_type(
            Inv.H1_G1,
            ConnectionType.AUX,
            RegisterType.HOLDING,
            special_registers=H1_AC1_REGISTERS,
        )
```

And then restart home assistant.

When it starts can you check in devices & services to see what entities are showing, and also report any foxess_modbus log entries

**Comment by markheppell** [comment]
[home-assistant_foxess_modbus_2024-12-28T10-48-58.661Z.log](https://github.com/user-attachments/files/18265888/home-assistant_foxess_modbus_2024-12-28T10-48-58.661Z.log)

There is now a lot of Not Provided entries, Connection Status has changed to Connected, Max Charge Current, Max Discharge current, Min SOC and Work Mode are showing values, all the rest are either Not Provided or Unknown

**Reply by FozzieUK** [reply]
ok, it's a start as the AC1 register range is being restricted.

Can you screen shot the foxess_modbus device page so I can see what entities are showing.

**Comment by markheppell** [comment]
![Modbus6](https://github.com/user-attachments/assets/e53cd6f0-15d8-40b3-a6b2-8b5ea1e25f54)

![Modbus5](https://github.com/user-attachments/assets/dc411c10-3ed7-4fab-821c-8fdec39546f8)

**Comment by FozzieUK** [comment]
Ok, right a lot of restrictions because of the old AC1's input registers - 

Again can you open file editor, open the file `/homeassistant/custom_components/foxess_modbus/inverter_profiles.py`

go to line 222 and edit it change special_registers to H1_G2_REGISTERS

i.e.

```
        InverterModelProfile(InverterModel.AC1, r"^AC1-([\d\.]+)", capacity_parser=CapacityParser.H1)
        .add_connection_type(
            Inv.H1_G1,
            ConnectionType.AUX,
            RegisterType.HOLDING,
            special_registers=H1_AC1_REGISTERS,
        )
```

becomes

```
        InverterModelProfile(InverterModel.AC1, r"^AC1-([\d\.]+)", capacity_parser=CapacityParser.H1)
        .add_connection_type(
            Inv.H1_G1,
            ConnectionType.AUX,
            RegisterType.HOLDING,
            special_registers=H1_G2_REGISTERS,
        )
```

Restart home assistant and when it restarts have a look at the devices & services page, and attach any modbus logs

**Comment by markheppell** [comment]
Now we're looking a bit healthier.

![Modbus7](https://github.com/user-attachments/assets/3a7adf4f-61da-49fa-a0c7-8581a4b8ff68)

[home-assistant_foxess_modbus_2024-12-28T11-38-18.554Z.log](https://github.com/user-attachments/files/18266032/home-assistant_foxess_modbus_2024-12-28T11-38-18.554Z.log)

**Reply by FozzieUK** [reply]
ok, thanks, looks much better - the Time periods and BMS low / high cell volts/temps i'd expect not to be available as they are not yet supported on the G2 (possibly soon).

I'll leave you to have a look through the sensors and make sure they are all reporting correct values (and scale) - i've got to pop out now for the afternoon, but i'll take a look at the logs later and see if I can fix any gaps.

**Reply by markheppell** [reply]
Thank you so much, I'll keep an eye on things :-)

**Reply by markheppell** [reply]
It all seems fine to me.  This is so good, after months of just getting data every 5-10 mins and the frequent drop outs of data, to have live data is incredible.  Many thanks once again.

**Reply by FozzieUK** [reply]
Ok great, I think there are a couple of things to fix but i’ll need to work through all the registers to produce a list of things to check and i’ll come back to you and if you could help with a bit more testing please.
Once we have it all working, i’ll need to create a type for the G2 AC1 and then add it properly to the code.

**Reply by markheppell** [reply]
Sure, I’ll be around to help with any testing, one thing I have noticed is the HA server is working considerably harder, processor is up from about 3% to about 50% and memory up from about 25% to 70% since we got this working.

![IMG_0238](https://github.com/user-attachments/assets/dedd2ae5-57df-4ff5-9ec6-f4646f78c810)

**Reply by markheppell** [reply]
Actually ignore that performance stuff, rebooted it and it’s settled right back down.

**Reply by FozzieUK** [reply]
It will depend on your hardware, I used to see 30-40% cpu on a raspPi 4 but I was polling at 6 second intervals - the integration default is poll every 10 seconds and if you have the default home assistant commit_interval it will be processing a lot more data to disc. It’s best to disable any entities you don’t need or aren’t applicable such as the PV1 and PV2 power, volts, current.

**Reply by FozzieUK** [reply]
Just been working through all the entities, the majority are looking good, there are a few that would respond better if we change to an H1(G2) type for register lookups (such as inverter state which is showing as off-grid as it uses the H1G1 lookup).

The master/manager versions are not showing correctly as Fox have changed the encoding for all new inverters, hopefully that one will be sorted out in a future release.

I think we can get the BMS kwh remaining so when you get chance could you do a test on the following register please and see what response you get.

37632, count of 1, holding

**Reply by markheppell** [reply]
values:
  "37632": 1035

**Reply by FozzieUK** [reply]
ok, thanks - is it fair to say your battery pack is at about 10.35kWh when fully charged and you take into account their state of health.

**Reply by markheppell** [reply]
Yes 2 EP5 batteries and the fox app says 10.36 kWh

**Reply by FozzieUK** [reply]
ok that's good enough for me.

If you don't mind another quick test could you - as before open the file `/homeassistant/custom_components/foxess_modbus/inverter_profiles.py`

go to line 219 and edit it change the type to Inv.H1_G2

i.e.

```
        InverterModelProfile(InverterModel.AC1, r"^AC1-([\d\.]+)", capacity_parser=CapacityParser.H1)
        .add_connection_type(
            Inv.H1_G1,
            ConnectionType.AUX,
            RegisterType.HOLDING,
            special_registers=H1_G2_REGISTERS,
        )
```

so it becomes

```
        InverterModelProfile(InverterModel.AC1, r"^AC1-([\d\.]+)", capacity_parser=CapacityParser.H1)
        .add_connection_type(
            Inv.H1_G2,
            ConnectionType.AUX,
            RegisterType.HOLDING,
            special_registers=H1_G2_REGISTERS,
        )
```

Restart home assistant and when it restarts have a look at the devices & services page - hopefully inverter state will be correct 'on-grid' and you will have a battery kwh remaining sensor.
Would you mind screen shotting the entities page again so I can see what is and isn't showing correctly please.

**Reply by markheppell** [reply]
![Modbus8](https://github.com/user-attachments/assets/104fa1ce-2fa9-462e-8a02-463f7e3232ef)

**Reply by FozzieUK** [reply]
ok, thanks - looking better.

I'll take a good look through all the sensors in the morning to cross check.

**Reply by FozzieUK** [reply]
adding AC1 G2 profile type in this PR https://github.com/nathanmarlor/foxess_modbus/pull/730

**Comment by markheppell** [comment]
Excellent, let me know if you want any testing or confirmation. I did your last update and reapplied our manual changes and it stayed working.
________________________________

**Comment by zeclab** [comment]
Hi,
I've just got the AC1 G2 and have the same issue. I tried version v1.13.1b2 as I noticed your fix was in there but still didn't work for me.
I think there is issue with the inverter name as had to comment out 274-280 and change line 267 from:
`InverterModel.AC1, r"^AC1-([\d\.]+)-E-G2", capacity_parser=CapacityParser.H1`
To:
`InverterModel.AC1, r"^AC1-([\d\.]+)", capacity_parser=CapacityParser.H1`
Then all the sensors populated after a restart.
The integration shows the device model as "AC1-5.0-E-G2 - AUX".
Let me know if you like me to test anything.

**Reply by FozzieUK** [reply]
Thanks, looks like another 'subversion' of H1 G2 to test for - 

If you could go to Developer Tools, Actions and select the action foxess modbus: read registers then see what the response is -

Start Address: 30000, Count: 16, Holding

And if you could post the response here.

**Reply by FozzieUK** [reply]
ok thanks, I can see why removing 274-280 would work, but looking at the inverter name it should really have worked without you having to change line 267 - when you've got a spare few minutes would you mind changing that back to what it was `InverterModel.AC1, r"^AC1-([\d\.]+)-E-G2", capacity_parser=CapacityParser.H1` and restart and check to see if that is ok.

**Reply by zeclab** [reply]
Ok so I changed it back and I'm getting the error again:
![image](https://github.com/user-attachments/assets/cf417824-6dd9-4655-8e07-34631635138b)

Here is what I am getting back from the debug log:

> 2025-01-30 13:01:55.108 DEBUG (MainThread) [custom_components.foxess_modbus.client.modbus_client] Closing connection to modbus on tcp://192.168.123.252:502
2025-01-30 13:02:05.110 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Reading addresses on tcp://192.168.123.252:502 247: (10016, 3)
2025-01-30 13:02:05.111 DEBUG (SyncWorker_5) [custom_components.foxess_modbus.client.custom_modbus_tcp_client] Connecting to CommParams(comm_name='comm', comm_type=<CommType.TCP: 1>, reconnect_delay=0.1, reconnect_delay_max=300, timeout_connect=3, host='192.168.123.252', port=502, source_address=None, handle_local_echo=False, sslctx=None, baudrate=-1, bytesize=-1, parity='', stopbits=-1)
2025-01-30 13:02:05.202 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Modbus error when polling tcp://192.168.123.252:502 247: Exception Response(132, 4, IllegalAddress)
2025-01-30 13:02:15.111 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Reading addresses on tcp://192.168.123.252:502 247: (10016, 3)
2025-01-30 13:02:15.190 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Modbus error when polling tcp://192.168.123.252:502 247: Exception Response(132, 4, IllegalAddress)
2025-01-30 13:02:25.112 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Reading addresses on tcp://192.168.123.252:502 247: (10016, 3)
2025-01-30 13:02:25.190 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Modbus error when polling tcp://192.168.123.252:502 247: Exception Response(132, 4, IllegalAddress)
2025-01-30 13:02:35.113 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Reading addresses on tcp://192.168.123.252:502 247: (10016, 3)
2025-01-30 13:02:35.206 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Modbus error when polling tcp://192.168.123.252:502 247: Exception Response(132, 4, IllegalAddress)
2025-01-30 13:02:45.115 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Reading addresses on tcp://192.168.123.252:502 247: (10016, 3)
2025-01-30 13:02:45.193 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Modbus error when polling tcp://192.168.123.252:502 247: Exception Response(132, 4, IllegalAddress)
2025-01-30 13:02:45.193 WARNING (MainThread) [custom_components.foxess_modbus.modbus_controller] tcp://192.168.123.252:502 247 - 5 failed poll attempts: now not connected. Last error: Error reading registers. Type: RegisterType.INPUT; start: 10016; count: 3; slave: 247 from tcp://192.168.123.252:502: Exception Response(132, 4, IllegalAddress)
2025-01-30 13:02:55.115 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Reading addresses on tcp://192.168.123.252:502 247: (10016, 3)
2025-01-30 13:02:55.194 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Modbus error when polling tcp://192.168.123.252:502 247: Exception Response(132, 4, IllegalAddress)
2025-01-30 13:03:05.116 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Reading addresses on tcp://192.168.123.252:502 247: (10016, 3)
2025-01-30 13:03:05.194 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Modbus error when polling tcp://192.168.123.252:502 247: Exception Response(132, 4, IllegalAddress)
2025-01-30 13:03:15.117 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Reading addresses on tcp://192.168.123.252:502 247: (10016, 3)
2025-01-30 13:03:15.194 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Modbus error when polling tcp://192.168.123.252:502 247: Exception Response(132, 4, IllegalAddress)
2025-01-30 13:03:25.118 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Reading addresses on tcp://192.168.123.252:502 247: (10016, 3)
2025-01-30 13:03:25.197 DEBUG (MainThread) [custom_components.foxess_modbus.modbus_controller] Modbus error when polling tcp://192.168.123.252:502 247: Exception Response(132, 4, IllegalAddress)

**Reply by FozzieUK** [reply]
ok finally I see the problem!, that's a new one.

The model number isn't `AC1-5.0-E-G2` it is `AC1-5.0-E-G2    ` observation test here ~ note there are 4 extra spaces at the end of the model number in the second example - you can see them below in the model number response at 30006/7 8224, 8224 i.e. hex 2020, 2020  which translates to 4 spaces.

In line 267 could you make the change to add 4 spaces to the end after G2 like this -

`InverterModel.AC1, r"^AC1-([\d\.]+)-E-G2    ", capacity_parser=CapacityParser.H1` 

and then restart and try again please.

**Reply by canton7** [reply]
We should be stripping spaces here: https://github.com/nathanmarlor/foxess_modbus/blob/main/custom_components/foxess_modbus/modbus_controller.py#L529. That shouldn't be an issue.

**Reply by canton7** [reply]
Note that those regexes are only evaluated on first connection (during the setup flow). Once you've successfully connected to the inverter once, we won't query the model again. 

If your inverter model has changed, you'll need to remove and re-add it in HA.

**Reply by zeclab** [reply]
Hi Canton7,

It's a brand new device so the model hasn't changed. I am removing/deleting the device and then adding again, between every test and as well as restarting HA.

Fozzie,

Copying and pasting what you gave didn't work either. My regex (and python) is pretty shocking but I've tried the below without success:
`InverterModel.AC1, r"^AC1-([\d\.]+)-E-G2\s+$"`
`InverterModel.AC1, r"^AC1-([\d\.]+)-E-G2\s*$"`
`InverterModel.AC1, r"^AC1-([\d\.]+)-E-G2(\s*$)"`
`InverterModel.AC1, r"^AC1-([\d\.]+)-E-G2([\d\.]+)"`
`InverterModel.AC1, r"^AC1-([\d\.]+)-E-G2 - AUX"` (for kicks)

Cheers

**Reply by FozzieUK** [reply]
@zeclab The original regex was correct if the trailing spaces are already stripped.. - I’d like to try something else can you reload the v1.13.1b2 release - but only change line 271 in inverter_profiles to this

versions={Version(1, 44): Inv.H1_G2_PRE144, None: Inv.H1_G2_SET},

Re-add your device and restart.

**Reply by zeclab** [reply]
@FozzieUK ah ok, I understand now.
I tried it but still no joy :(

> ➜  ~ ls /homeassistant/custom_components/foxess_modbus
__init__.py                common                     flow                       manifest.json              select.py                  translations
__pycache__                config_flow.py             icons.json                 modbus_controller.py       sensor.py
binary_sensor.py           const.py                   inverter_adapters.py       number.py                  services
client                     entities                   inverter_profiles.py       remote_control_manager.py  services.yaml
➜  ~ ls /tmp/foxess_modbus-1.13.1b2/custom_components/foxess_modbus 
__init__.py                config_flow.py             icons.json                 modbus_controller.py       sensor.py
binary_sensor.py           const.py                   inverter_adapters.py       number.py                  services
client                     entities                   inverter_profiles.py       remote_control_manager.py  services.yaml
common                     flow                       manifest.json              select.py                  translations
➜  ~ rm -rf /homeassistant/custom_components/foxess_modbus     
➜  ~ ls /homeassistant/custom_components/foxess_modbus             
ls: /homeassistant/custom_components/foxess_modbus: No such file or directory
➜  ~ cp -R /tmp/foxess_modbus-1.13.1b2/custom_components/foxess_modbus /homeassistant/custom_components 
➜  ~ ls /homeassistant/custom_components/foxess_modbus                                                 
__init__.py                config_flow.py             icons.json                 modbus_controller.py       sensor.py
binary_sensor.py           const.py                   inverter_adapters.py       number.py                  services
client                     entities                   inverter_profiles.py       remote_control_manager.py  services.yaml
common                     flow                       manifest.json              select.py                  translations

`        InverterModelProfile(`
`            InverterModel.AC1, r"^AC1-([\d\.]+)-E-G2", capacity_parser=CapacityParser.H1`
`        ).add_connection_type(`
`            ConnectionType.AUX,`
`           RegisterType.HOLDING,`
`            versions={Version(1, 44): Inv.H1_G2_PRE144, None: Inv.H1_G2_SET},`
`            special_registers=H1_G2_REGISTERS,`
`        ),`

Is there anything else you'd like me to try?

**Reply by FozzieUK** [reply]
I'm scratching my head a bit, can you enable debugging logging, then remove the device and re-add it again - you should get a log entry when it starts up that shows the autodetected inverter model and extra info -  can you post that please.

**Reply by canton7** [reply]
I'm really perplexed by this as well -- it defies all logic.

> I think there is issue with the inverter name as had to comment out 274-280 and change line 267 from:
InverterModel.AC1, r"^AC1-([\d\.]+)-E-G2", capacity_parser=CapacityParser.H1
To:
InverterModel.AC1, r"^AC1-([\d\.]+)", capacity_parser=CapacityParser.H1
Then all the sensors populated after a restart.

Can you post your full file after you edited it, and after confirming that your edits fixed foxess_modbus to the point where all sensors were updated correctly and there were no errors or warnings in the log?

**Reply by canton7** [reply]
Argh, think I've got it.

**Reply by canton7** [reply, marked as answer]
#750

**Comment by zeclab** [comment]
Thanks for the response.

Here is the output you requested:
values:
  "30000": 16707
  "30001": 12589
  "30002": 13614
  "30003": 12333
  "30004": 17709
  "30005": 18226
  "30006": 8224
  "30007": 8224
  "30008": 0
  "30009": 0
  "30010": 0
  "30011": 0
  "30012": 0
  "30013": 0
  "30014": 0
  "30015": 0

**Reply by canton7** [reply]
Can you confirm that this was definitely reading holding registers, not input registers?

**Comment by zeclab** [comment]
Yup that fixed it for me.

Thanks both!

**Reply by canton7** [reply]
Thanks for confirming. I'll get that released soon.

**Comment by markheppell** [comment]
Sorry to be a pain, but upgrading to the latest version has broken it again

![image](https://github.com/user-attachments/assets/0e363919-ae8f-4c65-8770-23ea87d1cb26)

**Reply by canton7** [reply]
What was your manager version?

**Reply by markheppell** [reply]
Master 1.31, Slave 1.02, ARM 1.48.  Is that what you mean?

**Reply by canton7** [reply]
OK, so you're on a recent version. Make sure that version is selected in the settings (under "CONFIGURE"). If that's correct, can you post your log?

I assume you're on v1.13.4?

**Reply by markheppell** [reply]
Okay, I think that's what I may be missing. CONFIGURE doesn't seem to give me that
![image](https://github.com/user-attachments/assets/0d649927-023b-45c7-8fd1-a69c49f395af)
![image](https://github.com/user-attachments/assets/248674f5-2e62-4df0-ac95-132ba0f619e7)
![image](https://github.com/user-attachments/assets/305d5cc1-0b19-49ce-a800-8d9b16e35bbe)

**Reply by canton7** [reply]
OK, something's up: the AC1-E-G2 should have 2 versions available. Can you double-check which version of the integration you're on (HA now displays it on the integration's page, the same page that has the "CONFIGURE" button on)? When you say you've upgraded, can you remember which version you upgraded from (and I assume that was working fine)?

**Reply by markheppell** [reply]
Aha.  Problem solved.  If I went through the Network Settings and followed it through then voila, it changed to
![image](https://github.com/user-attachments/assets/5433b4c4-0714-437b-a8e6-07c97f36be19)

Then I could set the version, and it's all back running

**Reply by canton7** [reply]
I think the config must have been saying that it was an AC1 G1: the old broken version would have stored that in the config, and the AC1 G1 only has 1 version (so you wouldn't have seen the version selector).

As you're using a recent manager version it should have defaulted to the correct version for you, but glad it's up and running again!

**Reply by markheppell** [reply]
So am I, you've no idea how much I appreciate this integration
![image](https://github.com/user-attachments/assets/f26fb5bc-8212-4cc6-b569-ca2c9e33c4b0)
